src/mocking.sh

```diff
@@ -39,13 +39,12 @@ validate-args() {
 args="\$@"
 
 if [ "\${args}" != "${expected}" ]; then
-
-    cat <<OUT
+    cat > \$0_error <<OUT
 Unexpected invocation for command 'some-command':
 Got :      <"\${args}">
 Expected : <"${expected}">
 OUT
-    exit 1
+    exit 1 # Kept for historical reasons. Proper usage would be mock xyz --with-args "arg1 arg2" --and exit-code 1.
```

Comment:
no really, this exit 1's goal was to break the test in case of wrong arguments. since you check that afterward now you can take that off.

Reply:
Currently, it breaks scripts that invoke it when they fail because of the exit code or when they are running in set -e mode. Removing it changes the contract with the test subject.

Reply:
Oh of course, i thought the error file lookup was after a mock but it's at the end of the test.  You're right sir ;)